sycl/source/detail/jit_compiler.cpp

```diff
@@ -630,6 +630,10 @@ updatePromotedArgs(const ::jit_compiler::SYCLKernelInfo &FusedKernelInfo,
   }
 }
 
+template <typename T> static auto getView(const std::vector<T> &Vec) {
```

Comment:
Would it make sense to collect those helpers in one place (header)?

Reply:
You mean `getView`, `toArrayRef` and `toVector`, right? I think `toArrayRef` can't live together with the others because of the reliance of the LLVM headers. (NB: If I'm wrong and we can include those, we could just use `ArrayRef` directly.)

Reply:
From what I can tell, the helpers are only used in `.cpp` files. Could we not place them in a new header in `common` that is not exposed to the outside, but allows us to define the helpers in a single place and use across different source files within the JIT compiler?

As this header would be internal to the JIT compiler only, including LLVM components would not be problematic.

Reply:
I channeled my inner @victor-eds and found a template-based solution: `View` (dropped the `Vector`, should be specific enough as it is in the `jit_compiler` namespace) can now be constructed from anything that offers `data()` and `size()`, and there is a templated `to` method which can return any type that can be constructed from `begin()` and `end()` iterators.